src/qibolab/_core/instruments/qcs.py

```diff
@@ -0,0 +1,343 @@
+
+            elif isinstance(channel, DcChannel):
+
+                pulses = []
+                sweep_param_map = sweeper_pulse_map.get(pulse.id, {})
+
+                for pulse in sequence.channel(channel_id):
+                    if pulse.kind == "delay":
+                        duration = (
+                            sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            ),
+                        )
+                    elif pulse.kind == "pulse":
+                        qcs_pulse = qcs.DCWaveform(
+                            duration=sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            ),
+                            envelope=generate_qcs_envelope(pulse.envelope),
+                            amplitude=sweep_param_map.get("amplitude", pulse.amplitude),
+                        )
+                    else:
+                        raise ValueError("Unrecognized pulse type", pulse.kind)
+
+                program.add_waveform(pulses, virtual_channel)
+        if hw_demod:
+            program = program.n_shots(num_shots)
+        return program
+
+    def play(
+        self,
+        configs: dict[str, Config],
+        sequences: list[PulseSequence],
+        options: ExecutionParameters,
+        sweepers: list[ParallelSweepers],
+    ) -> dict[int, Result]:
+
+        if options.relaxation_time is not None:
+            self.backend._init_time = int(options.relaxation_time)
+
+        ret: dict[PulseId, np.ndarray] = {}
+        for sequence in sequences:
+            results = self.backend.apply(
+                self.create_program(
+                    sequence.align_to_delays(), configs, sweepers, options.nshots
+                )
+            ).results
+            acquisitions = sequence.acquisitions
+            acquisition_map: dict[qcs.Channels, list[InputOps]] = {}
+
+            for channel_id, input_op in acquisitions:
+                channel = self.virtual_channel_map[channel_id]
+                if channel not in acquisition_map:
+                    acquisition_map[channel] = []
+                acquisition_map[channel].append(input_op)
+
+            averaging = options.averaging_mode is not AveragingMode.SINGLESHOT
+            for channel, input_ops in acquisition_map.items():
+                if options.acquisition_type is AcquisitionType.RAW:
+                    raw = results.get_trace(channel, averaging)
+                elif options.acquisition_type is AcquisitionType.INTEGRATION:
+                    raw = results.get_iq(channel, averaging)
+                elif options.acquisition_type is AcquisitionType.DISCRIMINATION:
+                    classifier = self.classifier_map[channel]
+                    raw = results.get_classified(channel, averaging, classifier)
+                else:
+                    raise ValueError("Acquisition type unrecognized")
```

Comment:
Optimal candidate for another method :)

Reply:
Implemented in https://github.com/qiboteam/qibolab/pull/944/commits/52270c4338f90ac3e42ab5ddc0f272afb0f92d67